packages/containers/src/Action/Action.connect.js

```diff
@@ -1,41 +1,31 @@
 import { api, cmfConnect } from '@talend/react-cmf';
 import { Action } from '@talend/react-components';
 
-import ActionButton from '../ActionButton';
-import ActionFile from '../ActionFile';
-import ActionSplitDropdown from '../ActionSplitDropdown';
-import ActionDropdown from '../ActionDropdown';
-import ActionIconToggle from '../ActionIconToggle';
-import getRenderers from '../renderers';
-
-export const renderers = {
-	ActionButton,
-	ActionFile,
-	ActionSplitDropdown,
-	ActionDropdown,
-	ActionIconToggle,
-};
-
 export function mapStateToProps(state, ownProps) {
-	const props = {
-		renderers: getRenderers(renderers),
-	};
+	const props = {};
+	if (ownProps.actionId) {
+		// eslint-disable-next-line no-console
+		console.warn(
+			'DEPRECATED: you should use componentId to map props using props.Action#componentId',
+		);
+		const info = api.action.getActionInfo(
+			{
+				registry: api.registry.getRegistry(),
+				store: {
+					getState: () => state,
+				},
+			},
+			ownProps.actionId,
+		);
 
-	if (!ownProps.actionId && !ownProps.name) {
-		return props;
+		props.displayMode = info.displayMode;
 	}
-	const info = api.action.getActionInfo(
-		{
-			registry: api.registry.getRegistry(),
-			store: {
-				getState: () => state,
-			},
-		},
-		ownProps.actionId || ownProps.name,
-	);
+	return props;
+}
 
-	props.actionId = ownProps.actionId || ownProps.name;
-	props.displayMode = info.displayMode;
+export function mergeProps(stateProps, dispatchProps, ownProps) {
+	const props = Object.assign({}, ownProps, stateProps, dispatchProps);
```

Comment:
beware Object.assign does a shallow merge, use https://lodash.com/docs/4.17.5#merge if it is intended to have a recursive merge